artemis/reporting/modules/vcs/reporter.py

```diff
@@ -0,0 +1,239 @@
+import os
```

Comment:
90% of this file isn't really the best. The "public repo" check is too naive. People often embed creds inside git repo URLs, since they are not logged into their git accounts. There shouldn't be a reason to perform a full git clone. That just wastes time.

Reply:
What do you mean with "People often embed creds inside git repo URLs, since they are not logged into their git accounts. " - what change would you expect? 

Regarding "There shouldn't be a reason to perform a full git clone. That just wastes time": we don't do full git clone - we try to use the available options (`--depth 1 --filter=blob:none --no-checkout`) to decrease the time (same with SVN). Do you have better ideas?

Reply:
Fair enough. Gotta start somewhere.